test/cli/smt2/test_smt2.t

```diff
@@ -111,3 +111,11 @@ Test uninterpreted:
   sat
   (model
     (x f32 0.))
+
+Test status tracking:
+  $ smtml run test_set_info_status.smt2
+  sat
+  smtml: internal error, uncaught exception:
+         Failure("Expected status: unsat, but solver returned sat")
```

Comment:
Can you maybe add a cli option, i.e. `--no-strict-status` that simply logs the error instead of using failwith? This is a breaking change so I want to allow people to have the option to disable this if they want.